Выполнение домашнего задания по Stream API

Sindicat007 opened the pull request.

Sindicat007 committed:
Выполнение домашнего задания по Stream API

Signed-off-by: Sindicat007 <sindicat007@ya.ru>

RinatBeybutov commented on src/StreamApi/task1/BasicStreamApi.java:
```diff
@@ -0,0 +1,67 @@
+        //Сформируйте список полных имен старше 30 лет.
+        List<String> peopleOver30 = people.stream()
+                .filter(person -> person.getAge() > 30)
+                .map(person1 -> person1.getFirstName() + " " + person1.getLastName())
```
здесь тоже person вместо person1

RinatBeybutov commented on src/StreamApi/task3/Main.java:
```diff
@@ -0,0 +1,43 @@
+        //Найдите сумму всех чисел, кратных 5.
+        List<Integer> multiply5 = numbers.stream()
+                .filter(number -> number % 5 == 0)
+                .toList();
```
а где сумма чисел?

RinatBeybutov commented on src/StreamApi/task4/Main.java:
```diff
@@ -0,0 +1,53 @@
+        String maxGradeStudent = students.stream()
+                .max(Comparator.comparingInt(Student::getGrade))
+                .map(person -> person.getName() + " " + person.getGrade())
+                .orElse(null);
```
лучше написать orElse("Такой студент отсутствует")

RinatBeybutov commented on src/StreamApi/task5/Main.java:
```diff
@@ -0,0 +1,34 @@
+
+        //Убедитесь, что все слова имеют длину не менее 5 символов.
+        List<String> moreThan5Char = list.stream()
+                .filter(s -> s.length() > 4)
```
здесь лучше применить операцию allMatch, а после фильтрации нужно сравнивать с изначальным списком

RinatBeybutov commented on src/StreamApi/task6/Main.java:
```diff
@@ -0,0 +1,79 @@
+        List<String> authorUnique = book.stream()
+                .map(Book::getAuthor)
+                .distinct()
+                .toList();
```
можно применить foreach в стримах и вывести в консоль сразу

RinatBeybutov commented on src/StreamApi/task6/Main.java:
```diff
@@ -0,0 +1,79 @@
+    public static void printPriceMore500(List<Book> book) {
+        List<Book> resultMore500 = book.stream()
+                .filter(s -> s.getPrice() > 500.00)
+                .toList();
```
можно применить foreach в стримах и вывести в консоль сразу

Sindicat007 committed:
task1:
Поменял название аргумента
task3:
Добавил подсчет суммы элементов потока
task4:
Отредактировал orElse в потоке поиска студента с наивысшей оценкой.

Signed-off-by: Sindicat007 <sindicat007@ya.ru>

Sindicat007 committed:
task5:
Изменил проверку на длину слов.
Добавил вывод в консоль информации о проверке листа строк, в поиске строк с максимальной длиной
task6:
Изменил способ печати в методах (printListWithUniqueAuthor, printPriceMore500)

Signed-off-by: Sindicat007 <sindicat007@ya.ru>
